packages/@react-aria/textfield/src/useTextField.ts

```diff
@@ -28,25 +36,34 @@ export interface TextFieldAria {
   errorMessageProps: HTMLAttributes<HTMLElement>
 }
 
-interface AriaTextFieldOptions extends AriaTextFieldProps {
+type TextFieldIntrinsicElementType = keyof Pick<IntrinsicHTMLElements, 'input' | 'textarea'>;
```

Comment:
I think I just may not know the definition well enough, but why Intrinsic?

Reply:
Right, they are in the environment, however, there is an `intrinsic` keyword in typescript 4.something https://gist.github.com/ruizb/55e1fc37cb198dccfdaf81450c3ebd43#intrinsic-type which means something else I think?
I think you may be using it though because it's a common naming pattern? if so, could you provide some background on it?

To share how my brain is working. I think about it more like, there are intrinsic attributes for any element type, such as an element's size, derived from contents, or the compiler (type engine) will recognize it as something it needs to apply special internal treatment to outside of the language (js) itself.

So when I read this line, what my brain says is:
`TextField ... Element Type naturally belonging to … and is either the type ‘input’ or ‘textarea’`

when what I think you're saying is
`TextField element type as either ‘input’ or ‘textarea’ from HTML`

maybe this is sufficient?
`type TextFieldElementType = keyof Pick<HTMLElementToInterfaceMap, 'input' | 'textarea'>`

While the comments were hugely helpful to explain IntrinsicHTMLElements, I had to keep going back to the comment to understand it. Can we name it something more along the lines of what it does?

Reply:
@snowystinger - I see where the confusion is. So, since the term "intrinsic" without any context means "native", if we take it in the context of TypeScript, it could refer to the [intrinsic string manipulation types](https://www.typescriptlang.org/docs/handbook/utility-types.html#intrinsic-string-manipulation-types) that are provided by TypeScript among other utility types.

Now, in the context of the HTML DOM, the term "intrinsic" here refers to the native HTML elements, such as `div`, `span`, `button`, `input`, etc. More specifically, the reason we're using this term here in our naming convention is to align with the same terminology used for the [`JSX.IntrinsicElements`](https://www.typescriptlang.org/docs/handbook/jsx.html#type-checking) interface.

With regards to naming it `TextFieldElementType` instead of `TextFieldIntrinsicElementType`, we can definitely do that. However, just to clarify, the idea of putting the term "intrinsic" in there was to distinguish the kind of element type it is; i.e. `input` vs. `HTMLInputElement`.

That said, if that's causing more confusion than clarity, I'd be more than happy to make the refactor.

Reply:
Ah, I'd forgotten about JSX.IntrinsicElements, thanks for the reminder. That definition does clear things up. In your comments, would you mind including a link to that?
I'm not sure that `IntrinsicElementType` though is correct following that definition because it's not actually the type, for intrinsic elements they are just their lower case string name. So maybe this would be better?
`type TextFieldIntrinsicElements = keyof Pick<IntrinsicElementsToInterfaceTypesMap, 'input' | 'textarea'>`

I appreciate your patience in this matter.